I double checked. `fstabfmt -i` makes `fstabfmt` panic instead of printing the usage information:

```
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
main.main()
	/home/afr/clones/fstabfmt-fsmiamoto/main.go:106 +0x70f
```

--- a/README.md
+++ b/README.md
@@ -7,12 +7,13 @@ Format `/etc/fstab` files.
 ## Features and limitations
 
 * Can format `/etc/fstab` files.
-* Will use 2 spaces between all fields, if they are of equal length.
+* Will use 2 spaces between all fields by default, if they are of equal length.
 * The shorter fields are padded with spaces.
 
 ## Example use
 
 * Run `fstabfmt /etc/fstab` to see the formatted output. No changes are made to the file.
+* Run `fstabfmt -s 8 /etc/fstab` to see the formatted output with 8 spaces between fields. Again, no changes are made to the file.
 * Run `fstabfmt -i /etc/fstab` to make changes to `/etc/fstab`. Always keep a backup, just in case.
 
 ## Example output

--- a/main.go
+++ b/main.go
@@ -12,7 +12,7 @@ const versionString = "fstabfmt 1.0.0"
 
 func usage() {
 	fmt.Println(versionString + `
-Usage: fstabfmt [-i FILE]
+Usage: fstabfmt [-i] [-s NUM] FILE
 
 fstabfmt formats /etc/fstab files.
 It can either read from stdin and print to stdout
@@ -21,7 +21,7 @@ or modify the given file if the -i flag is used.
 -h, --help         Display this help
 -v, --version      Display the current version
 -s, --spaces NUM   Specify the number of spaces used between fields 
--i FILE            Supply a file that will be modified
+-i                 If provided will save the changes to the file
 
 `)
 }

--- a/main.go
+++ b/main.go
@@ -2,10 +2,10 @@ package main
 
 import (
 	"bytes"
+	"flag"
 	"fmt"
 	"io/ioutil"
 	"os"
-	"strconv"
 )
 
 const versionString = "fstabfmt 1.0.0"
@@ -18,10 +18,10 @@ fstabfmt formats /etc/fstab files.
 It can either read from stdin and print to stdout
 or modify the given file if the -i flag is used.
 
--h, --help        Display this help
--v, --version     Display the current version
--s, --spaces NUM  Specify the number of spaces used between fields 
--i FILE           Supply a file that will be modified
+-h, --help         Display this help
+-v, --version      Display the current version
+-s, --spaces NUM   Specify the number of spaces used between fields 
+-i FILE            Supply a file that will be modified
 
 `)
 }
@@ -82,40 +82,30 @@ func format(data []byte, spaces int) []byte {
 
 func main() {
 	var (
-		data     []byte
-		err      error
-		filename = "-"
-		modify   bool
-		spaces   = 2
+		data        []byte
+		err         error
+		filename    = "-"
+		modify      bool
+		showVersion bool
+		spaces      = 2
 	)
 
-	skipNextArg := false
-	for i := range os.Args {
-		if skipNextArg {
-			skipNextArg = false
-			continue
-		}
-		switch os.Args[i] {
-		case "-v", "--version":
-			fmt.Println(versionString)
-			return
-		case "-h", "--help":
-			usage()
-			return
-		case "-i":
-			filename = os.Args[i+1]
-			modify = true
-			skipNextArg = true
-		case "-s", "--spaces":
-			spaces, err = strconv.Atoi(os.Args[i+1])
-			if err != nil {
-				fmt.Fprintln(os.Stderr, "error: invalid number of spaces")
-				os.Exit(1)
-			}
-			skipNextArg = true
-		default:
-			filename = os.Args[i]
-		}
+	flag.IntVar(&spaces, "s", 2, "")
+	flag.IntVar(&spaces, "spaces", 2, "")
+	flag.BoolVar(&showVersion, "v", false, "")
+	flag.BoolVar(&showVersion, "version", false, "")
+	flag.BoolVar(&modify, "i", false, "")
+	flag.Usage = usage
+
+	flag.Parse()
+
+	if showVersion {
+		fmt.Println(versionString)
+		os.Exit(0)
+	}
+
+	if flag.Arg(0) != "" {
+		filename = flag.Arg(0)
 	}
 
 	if filename == "-" {

